src/core_test/mpas_test_core_stream_list.F

```diff
@@ -0,0 +1,694 @@
+   !>
+   !> \param err  The error code that indicates the result of the test.
+   !
+   !--------------------------------------------------------------------------------------
+   subroutine mpas_test_insert_duplicate_at_begin(err)
+      integer, intent(out) :: err
+      type(MPAS_stream_list_type), pointer :: list
+      type(MPAS_stream_list_type), pointer :: s1
+      integer :: ierr
+
+      err = 0
+      call MPAS_stream_list_create(list)
+
+      allocate(s1)
+      s1%name = 'stream1'
+
+      call MPAS_stream_list_insert(list, s1, ierr)
+      if (ierr /= MPAS_STREAM_LIST_NOERR) then
+         err = err + 1
+      end if
+
+      call MPAS_stream_list_insert(list, s1, ierr)
+      if (ierr /= MPAS_STREAM_LIST_DUPLICATE) then
+         err = err + 1
+      end if
+      if (MPAS_stream_list_length(list) /= 1) then
+         err = err + 1
+      end if
+
+      call MPAS_stream_list_destroy(list)
+   end subroutine mpas_test_insert_duplicate_at_begin
+
+   !**************************************************************************************
+   !  Subroutine mpas_test_insert_duplicate_at_end
+   !
+   !> \brief   Test inserting a duplicate stream at the end of the list.
+   !>
+   !> \details This subroutine tests the insertion of a duplicate stream at the
+   !>          end of an MPAS stream list. It ensures that the correct error
+   !>          is returned when attempting to insert a duplicate stream.
+   !>
+   !> \param err  The error code that indicates the result of the test.
+   !
+   !--------------------------------------------------------------------------------------
+   subroutine mpas_test_insert_duplicate_at_end(err)
+      integer, intent(out) :: err
+      type(MPAS_stream_list_type), pointer :: list
+      type(MPAS_stream_list_type), pointer :: s1, s2
+      integer :: ierr
+
+      err = 0
+      call MPAS_stream_list_create(list)
+
+      allocate(s1)
+      s1%name = 'stream1'
+
+      allocate(s2)
+      s2%name = 'stream2'
+
+      call MPAS_stream_list_insert(list, s1, ierr)
+      if (ierr /= MPAS_STREAM_LIST_NOERR) then
+         err = err + 1
+      end if
+
+      call MPAS_stream_list_insert(list, s2, ierr)
+      if (ierr /= MPAS_STREAM_LIST_NOERR) then
+         err = err + 1
+      end if
+
+      call MPAS_stream_list_insert(list, s2, ierr)
+      if (ierr /= MPAS_STREAM_LIST_DUPLICATE) then
+         err = err + 1
+      end if
+      if (MPAS_stream_list_length(list) /= 2) then
+         err = err + 1
+      end if
+
+      call MPAS_stream_list_destroy(list)
+   end subroutine mpas_test_insert_duplicate_at_end
+
+   !**************************************************************************************
+   !  Subroutine mpas_test_insert_duplicate_pointer_with_different_name
+   !
+   !--------------------------------------------------------------------------------------
+   subroutine mpas_test_insert_duplicate_pointer_with_different_name(err)
```

Comment:
I'm not convinced of the value of this test, and so I'd suggest we don't add it.